cli/loader/src/lib.rs

```diff
@@ -379,8 +380,8 @@ impl Loader {
     pub fn new() -> Result<Self> {
         let parser_lib_path = match env::var("TREE_SITTER_LIBDIR") {
             Ok(path) => PathBuf::from(path),
-            _ => dirs::cache_dir()
-                .ok_or_else(|| anyhow!("Cannot determine cache directory"))?
+            _ => etcetera::base_strategy::choose_base_strategy()?
```

Comment:
```suggestion
            _ => etcetera::choose_base_strategy()?
```

Also, let's clean up the old cache files if this is running on macOS

Reply:
Do you think a warning message is helpful in this case?

Reply:
I think in this case it's fine to omit, as it's just cache files and nothing the user touched